bindings/python/gnucash_core.py

```diff
@@ -243,7 +243,8 @@ def __init__(self, book_uri=None, sessionOpenMode=None, instance=None):
                 # has to be loaded before it can be altered
                 # Any existing store obviously has to be loaded
                 # More background: https://bugs.gnucash.org/show_bug.cgi?id=726891
-                if book_uri[:3] != "xml" or not is_new:
+                is_new = 2
```

Comment:
It would be useful to have this as const or enum.

Reply:
It would be clearest to copy what I did in C++: 
```Python
is_new = (mode in (SessionOpenMode.SESSION_NEW_STORE, SessionOpenMode.SESSION_NEW_OVERWRITE)
if book_uri[:3] != "xml" or not is_new:
```
And as long as you're squashing magic numbers, slicing `book_uri` counts as one.

Reply:
@jralls I don't understand
> And as long as you're squashing magic numbers, slicing book_uri counts as one.

could you please explain?

Reply:
The `3` in `book_uri[:3]` is also a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants). To code it clearly you'd want to find the position of `:` in the URI and get the substring of that (or better call `gnc_uri_get_components()` or python's [`urllib.parse.parse`](https://docs.python.org/3/library/urllib.parse.html#url-parsing)) to set a variable `scheme` and test `scheme == 'xml'`.

OTOH it's not your code so you don't have to clean it up if you don't want to. If you do, there's more: The logic is a bit tortured: `if not (is_new and scheme == 'xml'):` would be clearer.

Reply:
I had thought of that too but as you said as it is not my code didn't want to take care of it yet. But the direction you point looks good and I'll implement that.